server/audit/utils.py

```diff
@@ -0,0 +1,55 @@
+import itertools as it
+from audit_template.models import AuditTemplate
+from audit.models import Audit
+from inventory_item.models import Item
+
+
+def fetch_inventory_items(template):
+    dict = {}
+    categories = ['Location', 'Plant', 'Zone', 'Aisle', 'Bin', 'Part_Number', 'Serial_Number']
+
+    # Constructing the dict based on what optional criteria were added
+    for filters, category in zip(
+            [template.location, template.plant, template.zones, template.aisles,
+             template.bins, template.part_number, template.serial_number],
+            categories):
+
+        if filters:
+            dict[category] = filters
+
+    sorted_keys = sorted(dict.keys())
+    all_categories = sorted(dict)
+
+    # Finding all the different combinations based on the criteria selected
+    combinations = it.product(*(dict[category] for category in all_categories))
+    queries = [x for x in combinations]  # converting the itertools object to a list
+
+    kwargs = {}
+    inventory_items_to_audit = []
+    for query in queries:
+        for category, value in zip(sorted_keys, query):
+            kwargs[category] = value
+
+        # Fetching the inventory items corresponding to the criteria selected
+        # for all combinations
+        inventory_items = Item.objects.filter(**kwargs)
+
+        # Appending the element of the Queryset object to the list
+        # (Didn't find a way to cast it to a list)
+        for i in range(len(inventory_items)):
+            inventory_items_to_audit.append(inventory_items[i]._id)
+
+    return inventory_items_to_audit
```

Comment:
Can we add a TODO here to look further into a more efficient solution?

Reply:
I have noted this down on our Discord server as to be looked into further.

Reply:
Done 👍